unified-installer/astra-unified-installer.sh

```diff
@@ -2552,43 +2619,28 @@ if trident_will_be_installed_or_modified; then
-
-                if ! components_include_connector && (components_include_trident || components_include_acp); then
-                    if prompt_user_yes_no "Would you like to upgrade Trident? Choosing no will exit the script (yes/no):"; then
-                        step_generate_torc_patch "$_EXISTING_TORC_NAME" "$(get_config_trident_image)" "" "" "$(get_config_trident_autosupport_image)"
-                        trident_operator_image_needs_upgraded && step_generate_trident_operator_patch
-                    else
-                        exit 0
-                    fi
-                fi
-
-                if components_include_connector; then
-                    if acp_is_enabled; then
-                        if prompt_user_yes_no "Would you like to upgrade Trident?"; then
-                            step_generate_torc_patch "$_EXISTING_TORC_NAME" "$(get_config_trident_image)" "" "" "$(get_config_trident_autosupport_image)"
-                            trident_operator_image_needs_upgraded && step_generate_trident_operator_patch
-                        else
-                            loginfo "Trident will not be upgraded"
-                        fi
-                    else
-                        if prompt_user_yes_no "Would you like to upgrade Trident? If you choose no, ACP will remain disabled"; then
-                            step_generate_torc_patch "$_EXISTING_TORC_NAME" "$(get_config_trident_image)" "" "" "$(get_config_trident_autosupport_image)"
-                            trident_operator_image_needs_upgraded && step_generate_trident_operator_patch
-                        else
-                            loginfo "Trident will not be upgraded and ACP will remain disabled."
-                        fi
-                    fi
+                    _warning_message="We cannot verify the version of the custom Trident image you provided"
+                    _warning_message+=" ($(get_config_trident_image). ACP support (23.10+) is not guaranteed."
+                    logwarn "$_warning_message"
+
+                    step_generate_torc_patch "${_generate_torc_args[@]}"
+                    trident_operator_image_needs_upgraded && step_generate_trident_operator_patch
+                elif prompt_user_yes_no "Would you like to upgrade Trident?"; then
+                    step_generate_torc_patch "${_generate_torc_args[@]}"
+                    trident_operator_image_needs_upgraded && step_generate_trident_operator_patch
+                else
+                    _msg="You have chosen to use a version of Trident that is not supported with the current version"
+                    _msg+=" of Astra Control. This may result in some App Data Management operations not functioning"
+                    _msg+=" correctly or being blocked within Astra Control. It is highly recommended to upgrade"
+                    _msg+=" Trident to ensure compatibility and proper functionality."
+                    logwarn "$_msg"
```

Comment:
Some of the Trident upgrade flow got condensed, while some of it was trimmed out since I realized it was being taken care of by code downstream (mainly the ACP-related deleted code below this comment).

Reply:
👍 I'll see about changing the messaging in the script as a whole, shouldn't be too terrible.